### PR TITLE
Add new dependencies for rocm variant for py-torch recipe

### PR DESCRIPTION
**Pull request by renjithravindrankannath**
Removed redundant path variables and updated cmake module path for ROCm 5.2

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-torch/package.py [hunk 1]
```diff
@@ -437,6 +437,10 @@ def enable_or_disable(variant, keyword="USE", var=None, newer=False):
             env.set("HIPCUB_PATH", self.spec["hipcub"].prefix)
             env.set("ROCTHRUST_PATH", self.spec["rocthrust"].prefix)
             env.set("ROCTRACER_PATH", self.spec["roctracer-dev"].prefix)
+            env.set("USE_NCCL", True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-torch/package.py]: This depends on the value of the `+nccl` variant and is already set below.

File: var/spack/repos/builtin/packages/py-torch/package.py [hunk 2]
```diff
@@ -437,6 +437,10 @@ def enable_or_disable(variant, keyword="USE", var=None, newer=False):
...
+            env.set("USE_PTHREADPOOL", True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-torch/package.py]: This is already set below for PyTorch 1.6+

Comment (renjithravindrankannath) [reply var/spack/repos/builtin/packages/py-torch/package.py]: Removed USE_NCCL.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-torch/package.py]: The above comment was for NCCL, this comment is for PTHREADPOOL

File: var/spack/repos/builtin/packages/py-torch/package.py (hunk 1 above)
Comment (renjithravindrankannath) [reply var/spack/repos/builtin/packages/py-torch/package.py]: Removed USE_NCCL.

File: var/spack/repos/builtin/packages/py-torch/package.py (hunk 2 above)
Comment (renjithravindrankannath) [reply var/spack/repos/builtin/packages/py-torch/package.py]: Removed PTHREADPOOL as well from the recipe.

**Review by srekolam — COMMENTED** [review]
LGTM

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**